satpy/composites/viirs.py

```diff
@@ -865,6 +865,24 @@ def _linear_normalization_from_0to1(
     data[mask] = data[mask] / theoretical_max
 
 
+def _check_moon_phase(datasets):
+    """Check if we have Moon phase as an input dataset and, if not, calculate it."""
+    if len(datasets) < 3 or len(datasets) > 4:
+        raise ValueError("Expected either 3 or 4 datasets, got %d" % (len(datasets),))
```

Comment:
I think it's a bit confusing that the argument checks is done in this function. I would rather have the calling function call eg `_check_three_or_four_arguments(datasets)`that just does the argument checking and then `moon_illum_fraction = get_moom_illum_fraction(dataset[3:4])`
(`dataset[3:4]`gives `[]`if there are only 3 elements in the list)

Reply:
This should be done now. Let me know what you think.